src/main/java/com/hyunec/cosmicbaseballinit/domain/baseball/model/service/BaseballServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.hyunec.cosmicbaseballinit.domain.baseball.model.service;
+
+import com.hyunec.cosmicbaseballinit.domain.baseball.model.Batting;
+import com.hyunec.cosmicbaseballinit.domain.baseball.model.BattingResult;
+import com.hyunec.cosmicbaseballinit.domain.baseball.model.PlateAppearances;
+import com.hyunec.cosmicbaseballinit.domain.baseball.model.dto.NewGameResponse;
+import com.hyunec.cosmicbaseballinit.domain.baseball.model.exception.ExceptionMessage;
+import com.hyunec.cosmicbaseballinit.domain.baseball.model.utils.generator.BattingGenerator;
+import java.time.LocalDateTime;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BaseballServiceImpl implements BaseballService {
+
+  private static final String SUCCESS_NEW_GAME = "다음 게임을 시작합니다";
+  private static final boolean END_GAME = true;
+
+  private final PlateAppearances plateAppearances = new PlateAppearances();
+  private final BattingGenerator battingGenerator;
+
+  public BaseballServiceImpl(
+      @Qualifier("randomBattingGenerator") final BattingGenerator battingGenerator) {
+    this.battingGenerator = battingGenerator;
+  }
+
+  @Override
+  public void batting() {
+    if (!checkFourBallOrOut()) {
+      Batting generateValue = battingGenerator.generator();
+      plateAppearances.batting(generateValue);
+    }
+  }
```

Comment:
```suggestion
  @Override
  public void batting() {
    if (checkFourBallOrOut()) {
      return;
    }
    
    Batting generateValue = battingGenerator.generator();
    plateAppearances.batting(generateValue);
  }
```

`nit` 저는 이런 방식을 더 선호하는 편 입니다.

Reply:
확실히 `!` 가 붙은 것을 꼼꼼히 보는것 보단, 이 방법이 깔끔한것 같습니다.

앞으로는 이런 방향으로 나아가는것이 좋다고 저도 생각합니다 리뷰 감사드립니다 :)